### ISSUE TITLE
Macro was defined in different places?

### ISSUE THREAD

**Issue by jeffythedragonslayer** [cc65/cc65#1778]
The "Note: Macro was defined here" message reports different line numbers sometimes.  I think it should report the line that .macro is on.

With broken_macro.s:
```
.macro blah
blah2
blah2
blah2
.endmacro

blah
```

ca65 .\broken_macro.s output:

```
.\broken_macro.s:2: Note: Macro was defined here
.\broken_macro.s:7: Error: Symbol 'blah2' is already defined
.\broken_macro.s:3: Note: Macro was defined here
.\broken_macro.s:7: Error: ':' expected
.\broken_macro.s:3: Note: Macro was defined here
.\broken_macro.s:7: Error: Symbol '.size' is already defined
.\broken_macro.s:3: Note: Macro was defined here
.\broken_macro.s:7: Error: Symbol 'blah2' is already defined
.\broken_macro.s:4: Note: Macro was defined here
.\broken_macro.s:7: Error: ':' expected
.\broken_macro.s:4: Note: Macro was defined here
.\broken_macro.s:7: Error: Symbol '.size' is already defined
.\broken_macro.s:4: Note: Macro was defined here        
```

**Comment by bbbradsmith** [comment]
When there's a macro error it reports the line where the macro was put to use (in this case line 7) and it also reports the definition line of the macro that generated the error (in this case lines 2 3 and 4 are all generating their own errors).

So, the given error locations are correct, even though it's confusing. (The macro system is pretty confusing overall.)

I agree the error could be improved, though I don't have any implementation suggestion offhand. I don't see a bug here, though, it's reporting all of the relevant error lines correctly.

**Comment by mrdudz** [comment]
I actually agree with the OP - the "Macro was defined here" should always point to the line the macro was started (ie the line containing .macro), not some random line inside of the macro.

Perhaps we can change the wording of the error message to make more clear it is pointing to a line inside the macro? I don't have a good idea on what that would be right now though :)

**Comment by bbbradsmith** [comment]
I don't agree. The line isn't random, it's the line that produced the error. That's really important. If you have a 100 line macro and the error is on line 50, you need to know about line 50, not line 1 where it says ".macro".

The wording isn't great, I agree with that much, but the actual line numbers chosen aren't random or wrong. They're the 2 relevant lines to the error:
* The line of the code that invoked the macro.
* The line of the macro source that produced the problem.

The line with ".macro" on it isn't an error. It's not the line that needs to be implicated.

It could help if the "Note: Macro was defined here" error more clearly indicated that it's a commentary on the directly preceding error. OP's example output is actually missing the first line that ca65 reports, which might be making it look less consistent about this than it is. It should have read:
```
.\broken_macro.s:7: Error: ':' expected
.\broken_macro.s:2: Note: Macro was defined here
.\broken_macro.s:7: Error: Symbol 'blah2' is already defined
.\broken_macro.s:3: Note: Macro was defined here
.\broken_macro.s:7: Error: ':' expected
.\broken_macro.s:3: Note: Macro was defined here
.\broken_macro.s:7: Error: Symbol '.size' is already defined
.\broken_macro.s:3: Note: Macro was defined here
.\broken_macro.s:7: Error: Symbol 'blah2' is already defined
.\broken_macro.s:4: Note: Macro was defined here
.\broken_macro.s:7: Error: ':' expected
.\broken_macro.s:4: Note: Macro was defined here
.\broken_macro.s:7: Error: Symbol '.size' is already defined
.\broken_macro.s:4: Note: Macro was defined here   
```
Every error is a pair, and it's exactly the 2 relevant lines for that specific error.

So, perhaps "Note: Macro was defined here" is an ambiguous wording. Maybe there is something better to say. Maybe we could refer to the preceding message more clearly, indicated it with an ^ arrow or something? Maybe we could name the containing macro? Possible suggestion:
```
file:100: Error: ':' expected
file:76: Note: ^ Expanded from macro 'blah'
```

Otherwise the error reporting is consistent with what would happen without the macro. If you were to comment out .macro/.endmacro and the usage blah, it demonstrates that each of these errors would be produced by the indicated line of code already:
```
;.macro blah
blah2
blah2
blah2
;.endmacro

;blah
```
Produces:
```
.\broken_macro.s:2: Error: ':' expected
.\broken_macro.s:3: Error: Symbol 'blah2' is already defined
.\broken_macro.s:3: Error: ':' expected
.\broken_macro.s:3: Error: Symbol '.size' is already defined
.\broken_macro.s:3: Error: Symbol 'blah2' is already defined
.\broken_macro.s:4: Error: ':' expected
.\broken_macro.s:4: Error: Symbol '.size' is already defined
```

As an unrelated issue, I think the '.size' error is pretty confusing, though it's unrelated to the macro error question, that's just a confusing error ca65 already produces.

**Comment by mrdudz** [comment]
> I don't agree. The line isn't random, it's the line that produced the error. That's really important.

Yes, of course. But its not the line the macro was defined in, which is what the error message suggests :) So either the error message should be changed (preferred, because as you said, the actual line that contains the error is whats important to know) - Or it should contain the line that contains .macro. In an ideal world, it would tell us both of those :)
